articles/android-recyclerviews/index.md

```diff
@@ -0,0 +1,235 @@
+4. *control* - recyclerviews allow the developer to have full control of the list items and their behavior. ListViews limited this.
+
+This article goes through how to create RecyclerViews. It will address the requirements and classes involved to develop a functioning RecyclerView. To comfortably follow through with this tutorial, you need:
+
+- **Android Studio** installed on your machine.
+- A basic understanding of **Android development** using **Kotlin** programming language
+
+##### Components of a RecyclerView
+First, let's go through the parts of a RecyclerView.
+
+1. ***RecyclerView***: this is the list view declared in your activity. It's where the list of data will be displayed.
+2. ***Layout Manager***: it defines how the list should organize our data. It could be horizontal, vertical, or a grid layout.
+3. ***Adapter***: this connects our data, usually a list, to our recyclerview. It also observes changes in the list and updates the recyclerView.
+4. ***ViewHolder***: this holds the View onto which we display the data.
+
+#### Writing the code
+Fire up your IDE and create a project with an Empty activity. We'll work in the `MainActivity` class. Once the project build is done, open the app-level `build.gradle` and add the following dependencies.
+
+```gradle
+//Glide
+kapt 'com.github.bumptech.glide:compiler:4.11.0'
+implementation 'com.github.bumptech.glide:glide:4.11.0'
+
+//RecyclerView
+implementation 'androidx.recyclerview:recyclerview:1.1.0'
+```
+
+We'll use Glide to load images to the `ImageViews`. Don't forget to add `apply plugin: 'kotlin-kapt'` to the top of the gradle file.
+
+Sync the project to load all dependencies.
+
+After gradle syncs, open the `activity_main.xml` file and add the code below.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:id="@+id/parent_view"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context=".MainActivity">
+
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:id="@+id/recyclerview"
+        android:layout_margin="10dp"
+        tools:listitem="@layout/list_item"
+        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"/>
+
+</LinearLayout>
+```
+
+This code adds the RecyclerView XML view. We give it an ID of `recyclerview` that we'll use as a reference in the activity. We also add `app:layoutManager` to set a layout manager to our recyclerView.
+
+We can do this using Kotlin as well. Since we need the basic Vertical ListView form, we use `LinearLayoutManager`. We have three layout managers that we can use. You can learn more about them in their documentation.
```

Comment:
I think it would be helpful to add a link here. 
👍 
"their documentation."

Reply:
Added the documentation link